File: instalacao-ci/application/models/Entity/Urna.java
package br.edu.unipampa.app.domain;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

@Entity(name = "urnas")
public class Urna {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
    private List<Voto> votosParaContagem = new ArrayList<>();

    public void recebe(Voto voto) {
        if (this.votosParaContagem == null) {
            this.votosParaContagem = new ArrayList<>();
        }

        this.votosParaContagem.add(voto);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public List<Voto> getVotosParaContagem() {
        return votosParaContagem;
    }

    public void setVotosParaContagem(List<Voto> votosParaContagem) {
        this.votosParaContagem = votosParaContagem;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Urna urna = (Urna) o;
        return Objects.equals(getId(), urna.getId()) &&
                Objects.equals(getVotosParaContagem(), urna.getVotosParaContagem());
    }

    @Override
    public int hashCode() {

        return Objects.hash(getId(), getVotosParaContagem());
    }

}

File: instalacao-ci/application/models/Entity/Urna.php
<?php

namespace Entity;
use Doctrine\Common\Collections\ArrayCollection;


/**
 * @Entity
 * @Table(name="urnas")
 */
class Urna
{

    /**
     * @Id
     * @Column(type="integer", nullable=false)
     * @GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @ManyToMany(targetEntity="Voto", orphanRemoval=true)
     * @JoinTable(name="urnas_votos",
     *      joinColumns={@JoinColumn(name="urna_id", referencedColumnName="id")},
     *      inverseJoinColumns={@JoinColumn(name="votos_id", referencedColumnName="id", unique=true)}
     *      )
     */
    private $votosParaContagem;

    /**
     * Urna constructor.
     */
    public function __construct()
    {
        $this->votosParaContagem = new ArrayCollection();
    }

    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param mixed $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return mixed
     */
    public function getVotosParaContagem()
    {
        return $this->votosParaContagem;
    }

    /**
     * @param mixed $votosParaContagem
     */
    public function setVotosParaContagem($votosParaContagem)
    {
        $this->votosParaContagem = $votosParaContagem;
    }

//    public void recebe(Voto voto) {
//        if (this.votosParaContagem == null) {
//            this.votosParaContagem = new ArrayList<>();
//        }
//
//        this.votosParaContagem.add(voto);
//    }

}
